src/app/parcels/batch/submitTableData.ts

```diff
@@ -0,0 +1,315 @@
+
+    return { clientId: clientId, error: null };
+};
+
+const batchParcelToParcelRecord = (
+    parcel: ParcelData,
+    clientId: string,
+    listType: ListType
+): ParcelDatabaseInsertRecord => {
+    const collectionInfo: CollectionInfo | null = parcel.collectionInfo;
+
+    if (collectionInfo) {
+        const collectionDateTime = mergeDateAndTime(
+            collectionInfo.collectionDate,
+            collectionInfo.collectionSlot
+        ).toISOString();
+
+        return {
+            client_id: clientId,
+            collection_centre: collectionInfo.collectionCentreId,
+            collection_datetime: collectionDateTime,
+            list_type: listType,
+            packing_date: parcel.packingDate,
+            packing_slot: parcel.packingSlot,
+            voucher_number: parcel.voucherNumber,
+        };
+    }
+
+    return {
+        client_id: clientId,
+        list_type: listType,
+        packing_date: parcel.packingDate,
+        packing_slot: parcel.packingSlot,
+        voucher_number: parcel.voucherNumber,
+    };
+};
+
+const submitParcelRowToDb = async (
+    parcelRecord: ParcelDatabaseInsertRecord
+): Promise<InsertParcelReturnType> => {
+    const { data, error } = await supabase
+        .from("parcels")
+        .insert(parcelRecord)
+        .select("primary_key")
+        .single();
+
+    if (error) {
+        const logId = await logErrorReturnLogId("Error with insert: parcel data", error);
+        return { parcelId: null, error: { type: "failedToInsertParcel", logId } };
+    }
+
+    return { parcelId: data.primary_key, error: null };
+};
+
+const checkForRequiredClientData = (client: BatchClient): boolean => {
+    const hasRequiredFields: boolean =
+        client.fullName !== null &&
+        client.address !== null &&
+        client.adultInfo !== null &&
+        client.childrenInfo !== null &&
+        client.listType !== null &&
+        client.babyProducts !== null &&
+        client.deliveryInstructions !== null &&
+        client.extraInformation !== null &&
+        client.attentionFlag !== null &&
+        client.signpostingCall !== null &&
+        client.signpostingCall !== null &&
+        client.attentionFlag !== null;
+
+    return hasRequiredFields;
+};
+
+const checkForRequiredParcelData = (parcel: ParcelData): boolean => {
+    const hasRequiredFields: boolean =
+        parcel.packingDate !== null &&
+        parcel.packingSlot !== null &&
+        parcel.shippingMethod !== null;
+
+    const validCollectionOrDelivery: boolean =
+        parcel.shippingMethod !== "Collection" ||
+        (parcel.shippingMethod === "Collection" && parcel.collectionInfo !== null);
+
+    return hasRequiredFields && validCollectionOrDelivery;
+};
+
+const checkForEmptyParcel = (parcel: ParcelData): boolean => {
+    return Object.values(parcel).every((value) => value == null);
+};
+
+const checkForEmptyRow = (data: BatchEditData): boolean => {
+    return data === emptyBatchEditData;
+};
```

Comment:
these functions could be renamed to be clearer in exactly what theyre checking - taking the last one as an example, maybe a name like checkRowIsNotEmpty might be more appropriate? the name checkForEmptyRow sounds to me like it's taking an array of rows and checking if it contains an empty row.